### PR TITLE
command: Unpack latest thunk type inplace instead of in local

### PR DESCRIPTION
**Pull request by 3noch**
I have:

  - [x] Based work on latest `develop` branch
  - [x] Looked for lint in my changes with `hlint .` (lint found code you did not write can be left alone)
  - [ ] Run the test suite: `$(nix-build -A selftest --no-out-link)`
  - [ ] (Optional) Run CI tests locally: `nix-build release.nix -A build.x86_64-linux --no-out-link` (or `x86_64-darwin` on macOS)

### CODE REVIEW

**Comment by 3noch** [comment]
Still one failing test in selftest...

File: lib/command/src/Obelisk/Command/Thunk.hs [hunk 1]
```diff
@@ -555,7 +541,7 @@ let fetch = { private ? false, fetchSubmodules ? false, owner, repo, rev, sha256
     inherit owner repo rev sha256 fetchSubmodules private;
   };
   json = builtins.fromJSON (builtins.readFile ./github.json);
-in if builtins.pathExists ./local then ./local else fetch json
+in fetch json
```
Comment (kmicklas) [inline lib/command/src/Obelisk/Command/Thunk.hs:544]: This causes `Thunk update: ReadThunkError_UnrecognizedThunk` when trying to update a thunk from before this change. I think `ThunkSpec`s need to be immutable.

**Comment by ryantrinkle** [comment]
@kmicklas I think the ThunkSpecs are meant to be immutable once released, but I don't think the one with `local` was actually released.  If it was, we should definitely add support for it.  I do think it's probably a bit tricky to try to maintain support for anything that was ever on develop, since we could find bugs in the ThunkSpec itself after that.

@3noch Any thoughts?

**Comment by 3noch** [comment]
@kmicklas Yes that format was never released. It made it to `develop` but never `master`. Sadly there are probably some of us that accidentally ended up with the broken intermediate version.

**Pull request merged**